qualtran/bloqs/qft/qft_phase_gradient.py

```diff
@@ -18,24 +18,23 @@
 import cirq
 from numpy.typing import NDArray
 
-from qualtran import GateWithRegisters, QFxp, QUInt, Signature
+from qualtran import bloq_example, BloqDocSpec, GateWithRegisters, QFxp, QUInt, Signature
 from qualtran.bloqs.arithmetic.multiplication import PlusEqualProduct
 
 
 @attrs.frozen
 class QFTPhaseGradient(GateWithRegisters):
-    r"""QFT implemented using phase gradient trick. Uses O(n**2) T-gates for an n-bit register.
-
-    Given an n-bit phase gradient state $|\phi\rangle$ prepared as
+    r"""QFT implemented using coherent addition into a phase gradient register
 
+     A variant of the Quantum Fourier Transform (QFT) that utilizes an additional register provided in a phase gradient state to switch controlled rotations to coherent additions. Given an ancilla register prepared in the state
     $$
-        |\phi\rangle = \frac{1}{\sqrt{2^{n}}} \sum_{k=0}^{2^{n} - 1} \omega_{n}^{-k} |k\rangle
+        \frac{1}{\sqrt{2^{n}}} \sum_{k=0}^{2^{n} - 1} \omega_{n}^{-k} |k\rangle,
     $$
+    then coherent addition from the system into the ancilla applies the same phase that the controlled rotation in textbook QFT does. This reduces the number of T-gates to O(`bitsize`^2) and requires no additional arbitrary rotations beyond the one time ancilla preparation cost.
```

Comment:
nit: "O(`bitsize`^2)" should probably be $\mathcal{O}(n^2)$ where `n` is the `bitsize`.